pallets/automation-time/src/benchmarking.rs

```diff
@@ -133,28 +130,33 @@ fn schedule_auto_compound_delegated_stake_tasks<T: Config>(
 		let provided_id = AutomationTime::<T>::generate_auto_compound_delegated_stake_provided_id(
 			&owner, &collator,
 		);
-		let task_id = AutomationTime::<T>::schedule_task(
-			owner.clone(),
-			provided_id.clone(),
-			vec![time.clone()],
-		)
-		.unwrap();
 		let frequency = 3600;
 		let account_minimum = T::Currency::minimum_balance().saturating_mul(ED_MULTIPLIER.into());
 		let task = TaskOf::<T>::create_auto_compound_delegated_stake_task(
 			owner.clone(),
-			provided_id,
+			provided_id.clone(),
 			time,
 			frequency,
 			collator,
 			account_minimum,
 		);
+		let task_id = AutomationTime::<T>::schedule_task(&task, provided_id, vec![time]).unwrap();
 		<AccountTasks<T>>::insert(owner.clone(), task_id, task);
 		tasks.push((task_id, Pallet::<T>::get_account_task(owner.clone(), task_id).unwrap()));
 	}
 	tasks
 }
 
+fn increment_provided_id(mut provided_id: Vec<u8>) -> Vec<u8> {
```

Comment:
Left over change from when I was benchmarking with thousands of tasks. Leaving it in because we'll need something like this when we go back to realistic slot capacity.